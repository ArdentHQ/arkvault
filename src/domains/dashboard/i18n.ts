export const translations = {
	DASHBOARD_PAGE: {
		TITLE: "Dashboard",
	},
	FILTER_WALLETS: {
		CRYPTOASSET: {
			DESCRIPTION: "Select the networks to show.",
			TITLE: "Filter Networks",
		},
		WALLETS: {
			DESCRIPTION: "Select which wallets to view.",
			TITLE: "Your Wallets",
		},
	},

	LATEST_TRANSACTIONS: {
		EMPTY_MESSAGE:
			"You don't have any transactions yet. Once transactions have been made they will show up in your history.",
		NO_RESULTS: "No <bold>{{type}}</bold> transactions could be found.",
		TITLE: "Latest Transactions",
	},

	PORTFOLIO_BREAKDOWN: {
		FILTERED: "Please enable at least one public network to display your portfolio report.",
	},

	PORTFOLIO_BREAKDOWN_DETAILS: {
		TITLE: "Portfolio Breakdown",
	},

	WALLET_CONTROLS: {
		CREATE: "Create",
		EMPTY_MESSAGE: "Your portfolio is currently empty. <1>Create</1> or <3>Import</3> a wallet to get started.",
		EMPTY_MESSAGE_FILTERED: "You don't have any wallets matching the selected networks.",
		EMPTY_MESSAGE_TYPE: "You don't have any <bold>{{type}}</bold> wallets.",
		EMPTY_MESSAGE_TYPE_FILTERED: "You don't have any <bold>{{type}}</bold> wallets matching the selected networks.",
		IMPORT: "Import",
		IMPORT_LEDGER: "Import Ledger",
	},
};
