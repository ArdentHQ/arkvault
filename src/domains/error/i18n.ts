export const translations = {
	APPLICATION: {
		DESCRIPTION: "It looks like something went wrong. Try visiting another page or reloading.",
		HELP_TEXT: "Please provide the error below when contacting support.",
		RELOAD: "Reload",
		TITLE: "Something went wrong",
	},
	OFFLINE: {
		DESCRIPTION: "It looks like there may be a problem connecting to the Internet. Check your connection.",
		TITLE: "Connectivity Issue",
	},
};
