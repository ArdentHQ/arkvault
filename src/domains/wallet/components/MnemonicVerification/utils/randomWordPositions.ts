import { sample } from "@ardenthq/sdk-helpers";

const randomWordPositions = (length: number): number[] => {
	const positions: number[] = [...Array.from({ length }).keys()];
	const result: number[] = [];

	while (result.length < 3) {
		const randomNumber = sample(positions) + 1;

		if (result.includes(randomNumber)) {
			continue;
		}

		result.push(randomNumber);
	}

	return result.sort((a, b) => a - b);
};

export { randomWordPositions };
