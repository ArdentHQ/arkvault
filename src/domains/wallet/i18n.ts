export const translations = {
	CONNECT_LEDGER: {
		DESCRIPTION: "Connect your Ledger device, open the ARK app, and authorize the connection with ARK Vault.",
		HEADER: "Connect and Authorize Ledger Device",
		TITLE: "Connect Ledger Device",
		WAITING_DEVICE: "Waiting for confirmation  ...",
	},

	MESSAGE: "Message",

	MNEMONIC_VERIFICATION: {
		SELECT_WORD: "Select the {{position}}{{ordinalIndicator}} Word",
		WORD_NUMBER: "The {{position}}{{ordinalIndicator}} Word",
		WORD_REQUIRED: "Word required",
		WRONG_WORD: "Incorrect word entered",
	},

	MODAL_DELETE_WALLET: {
		DESCRIPTION:
			"Are you sure you wish to delete the selected wallet? Before deleting, make sure the mnemonic passphrase is secured safely.",
		TITLE: "Delete Wallet",
	},

	// @TODO: Remove this once the new wallet ledger create flow is complete
	MODAL_LEDGER_WALLET: {
		COMPATIBILITY_ERROR:
			"ARK Vault requires the use of a chromium based browser when using a Ledger device. Please use another browser like Chrome.",
		CONNECT_DEVICE: "Connect your Ledger and confirm input.",
		CONNECT_DEVICE_MODEL: "Connect your {{model}} and confirm input.",
		CONNECT_SUCCESS: "Successfully connected",
		DEVICE_NOT_AVAILABLE: "Unable to detect Ledger device",
		DEVICE_NOT_SUPPORTED:
			"A <bold>{{connectedModel}}</bold> is connected, but his operation requires a <bold>{{supportedModel}}</bold>. Connect the appropriate device to continue.",
		GENERIC_CONNECTION_ERROR:
			"Unable to connect to Ledger device. Please ensure that all other applications that connect to your Ledger are closed.",
		LEDGER_NANO_S: "Ledger Nano S",
		LEDGER_NANO_X: "Ledger Nano X",
		NO_DEVICE_FOUND: "Ledger device has not been found in time",
		OPEN_APP: "Open the {{coin}} app on your device ...",
		TITLE: "Ledger Wallet",
		UPDATE_ERROR: "The {{coin}} app version is {{version}}. Please update the {{coin}} app via Ledger Live.",
		WAITING_DEVICE: "Waiting for Ledger ...",
	},

	MODAL_NAME_WALLET: {
		DESCRIPTION: "Enter a name for this wallet address (only visible to you).",
		TITLE: "Wallet Name",
	},

	MODAL_RECEIVE_FUNDS: {
		DESCRIPTION: "Specify and confirm the amount to request.",
		DOWNLOAD_QR_CODE: "Download QR-Code",
		SPECIFY_AMOUNT: "Specify Amount",
		TITLE: "Receive Funds",
		WARNING:
			"Please note that you have exceeded the number of characters allowed, anything over {{maxLength}} characters will not appear in memos.",
	},

	MODAL_SELECT_ACCOUNT: {
		DESCRIPTION: "Locate and select the address for receiving funds.",
		SEARCH_PLACEHOLDER: "Enter name or address",
		TITLE: "Select Account",
	},

	MODAL_WALLET_ENCRYPTION: {
		DESCRIPTION:
			"You are about to register a second passphrase while having your current {{importType}} encrypted with a custom password. Continuing with the second signature registration will automatically remove the encryption once the transaction is sent. Ensure your {{importType}} is backed up before proceeding. You will have to re-import your wallet if you want to use an encryption password again.",
		TITLE: "Wallet Encryption Removal",
	},

	MODAL_WALLET_UPDATE: {
		DESCRIPTION_1: "A new update has been released for your wallet. You can download or postpone the update.",
		DESCRIPTION_2: "The update has been successfully downloaded and is ready to be installed.",
		TITLE: "Wallet Update {{version}}",
	},

	PAGE_CREATE_WALLET: {
		NETWORK_STEP: {
			GENERATION_ERROR:
				"An error occurred while creating your new address, please try again. If the error persists, kindly get in touch with our support team.",
			SUBTITLE: "Select a cryptoasset to create your new wallet address.",
			TITLE: "Select a Cryptoasset",
		},

		PASSPHRASE_CONFIRMATION_STEP: {
			PASSPHRASE_DISCLAIMER: "I am aware that if I lose my passphrase, I will lose access to my funds.",
			SUBTITLE:
				"Confirm that youâ€™ve saved your secret passphrase by correctly entering the word in the designated input field below.",
			TITLE: "Confirm Your Passphrase",
		},

		PASSPHRASE_STEP: {
			COPY_OR_DOWNLOAD: {
				DESCRIPTION: "You can copy or download your mnemonic, but store it safely.",
				TITLE: "Copy or Download Mnemonic",
			},
			ENCRYPTION: {
				DESCRIPTION:
					"Set an encryption password to use in place of your mnemonic passphrase. Note that you must still record and keep your mnemonic passphrase safe as losing this will result in you losing all access to your funds.",
				TITLE: "Use Wallet Encryption",
			},
			TITLE: "Your Passphrase",
			WARNING:
				"You are responsible for storing and protecting this mnemonic passphrase offline. ARK Vault cannot reveal this to you at a later time. If you lose this mnemonic passphrase, you will lose your funds.",
		},

		PROCESS_COMPLETED_STEP: {
			SUBTITLE: "The wallet address has been successfully created.",
			TITLE: "Completed",
		},

		TITLE: "Create Wallet",
	},

	PAGE_IMPORT_WALLET: {
		CANCELLING_STATE: {
			TITLE: "Cancelling Ledger Import, please wait ...",
		},

		ENCRYPT_PASSWORD_STEP: {
			CONFIRM_PASSWORD_LABEL: "Confirm Encryption Password",
			PASSWORD_LABEL: "Encryption Password",
			TITLE: "Encryption Password",
			WARNING:
				"This password does not replace your mnemonic passphrase. You must ensure that your passphrase is properly recorded and backed up so that you do not lose access to your funds. You cannot use your encryption password to restore your profile, so it is vital that you have your mnemonic passphrase readily available should you run into any issues (such as losing access to your local profile or clearing your browser's cache).",
		},

		LEDGER_CONNECTION_STEP: {
			SUBTITLE: "Open the app on your Ledger and check for details.",
			TITLE: "Open App on Ledger",
		},

		LEDGER_IMPORT_STEP: {
			SUBTITLE_one: "Your Ledger address has been imported.",
			SUBTITLE_other: "Your Ledger addresses have been imported.",
			TITLE: "Completed",
		},

		LEDGER_SCAN_STEP: {
			ADD_NEW_ADDRESS: "Add New Address",
			LOADED_SINGLE_WALLET: "Loaded <strong>1</strong> Wallet",
			LOADED_WALLETS: "Loaded <strong>{{count}}</strong> Wallets",
			SHOW_ALL: "Show All ({{count}})",
			SUBTITLE: "Select the addresses that you want to import.",
			TITLE: "Addresses",
		},

		METHOD_STEP: {
			ENCRYPTION: {
				DESCRIPTION:
					"Set an encryption password to use in place of your mnemonic passphrase. Note that you must still record and keep your mnemonic passphrase safe as losing this will result in you losing all access to your funds.",
				NOT_AVAILABLE: "Encryption not available for this import type",
				TITLE: "Use Wallet Encryption",
			},
			MNEMONIC_TIP: {
				GUIDELINES_1: "Ensure all words are in lowercase.",
				GUIDELINES_2: "Insert a space between each word.",
				GUIDELINES_3: "Verify that all words are spelled correctly.",
				GUIDELINES_4:
					"If copying and pasting, ensure no extra whitespace is added at the end of the copied text.",
				GUIDELINES_TITLE: "Please follow these guidelines when inputting your mnemonic into ARK Vault:",
				TITLE: "Need help with importing?",
			},
			SUBTITLE: "Select the import method by which you want to import the address to your Profile.",
			TITLE: "Import Wallet",
			TYPE: "Import Type",
		},

		NETWORK_STEP: {
			SUBTITLE: "Select a cryptoasset to import your existing wallet address.",
			TITLE: "Select a Cryptoasset",
		},

		PASSPHRASE_CONFIRMATION_STEP: {
			SUBTITLE: "Confirm your password to continue",
			TITLE: "Confirm your passphrase",
		},

		SUCCESS_STEP: {
			SUBTITLE: "Wallet import is complete. Now you can use it.",
			TITLE: "Completed",
		},

		TITLE: "Import Wallet",

		VALIDATION: {
			DECRYPT_WIF_ASSERTION: "Failed to decrypt WIF. Please check your password.",
			INVALID_MNEMONIC: "The given value is not BIP39 compliant",
			INVALID_PRIVATE_KEY: "Invalid Private Key",
			INVALID_PUBLIC_KEY: "Invalid Public Key",
			INVALID_SECRET: "The given value is BIP39 compliant. Please change Import Type to 'Mnemonic'",
			INVALID_WIF: "Invalid WIF",
		},
	},

	PAGE_WALLET_DETAILS: {
		ADDITIONAL_OPTIONS: "Additional Options",
		COPY_ADDRESS: "Copy Address",
		COPY_PUBLIC_KEY: "Copy Public Key",
		OPTIONS: {
			DELETE: "Delete",
			MULTISIGNATURE: "Multisignature",
			RECEIVE_FUNDS: "Receive Funds",
			RECEIVE_FUNDS_QR: "QR",
			REGISTER_DELEGATE: "Delegate",
			RESIGN_DELEGATE: "Resign Delegate",
			SECOND_SIGNATURE: "Second Signature",
			SIGN_MESSAGE: "Sign Message",
			STORE_HASH: "Store Hash",
			TRANSACTION_HISTORY: "Transaction History",
			VERIFY_MESSAGE: "Verify Message",
			WALLET_NAME: "Wallet Name",
		},

		PENDING_TRANSACTIONS: "Pending Transactions",
		PRIMARY_OPTIONS: "Manage Address",
		REGISTRATION_OPTIONS: "Register",

		STARRED_FIRST: "Show starred wallets at the top",

		STAR_WALLET: "Star",

		UNSTAR_WALLET: "Unstar",

		VOTES: {
			ACTIVE_COUNT: "Active {{count}}",
			ACTIVE_one: "Active",
			ACTIVE_other: "All Active",
			DELEGATE_STATUS: "Delegate Status",
			EMPTY_DESCRIPTION: "You have not voted for a Delegate yet.",
			LOCKED_UNVOTES: "Locked Unvotes",
			LOCKED_VOTES: "Locked Votes",
			MULTIVOTE: "Multivote",
			NOT_FORGING_COUNT: "{{count}} of your delegates are currently not in a forging position",
			NOT_FORGING_one: "Your delegate is currently not in a forging position",
			NOT_FORGING_other: "Your delegates are currently not in a forging position",
			RESIGNED_COUNT: "Resigned {{count}}",
			RESIGNED_one: "Resigned",
			RESIGNED_other: "All Resigned",
			STANDBY_COUNT: "Standby {{count}}",
			STANDBY_one: "Standby",
			STANDBY_other: "All Standby",
			TITLE_one: "My Vote",
			TITLE_other: "My Votes",
			VOTING_FOR: "Voting for",
		},

		YOUR_WALLETS: "Your Wallets",
	},

	SIGNATURE: "Signature",

	STATUS: {
		ACTIVE: "Active",
		RESIGNED: "Resigned",
		STANDBY: "Standby",
	},

	UPDATE_WALLET_DATA: "Update Wallet Data",

	UPDATING_WALLET_DATA: "Updating Wallet Data",

	VALIDATION: {
		ALIAS_ASSIGNED: "The name '{{alias}}' is already assigned to another wallet",
	},

	WALLET_NAME: "Wallet Name",
};
