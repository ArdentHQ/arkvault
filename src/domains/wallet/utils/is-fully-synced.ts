import { Contracts } from "@payvo/sdk-profiles";

export const isFullySynced = (wallet: Contracts.IReadWriteWallet): boolean =>
	wallet.hasBeenFullyRestored() && (wallet.hasSyncedWithNetwork() || wallet.balance() === 0);
