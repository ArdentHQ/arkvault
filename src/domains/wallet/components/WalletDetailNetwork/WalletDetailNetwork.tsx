import { Networks } from "@payvo/sdk";
import React from "react";
import { useTranslation } from "react-i18next";

import { NetworkIcon } from "@/domains/network/components/NetworkIcon";
import { TransactionDetailProperties } from "@/domains/transaction/components/TransactionDetail/TransactionDetail";
import { WalletDetail } from "@/domains/wallet/components/WalletDetail";
import { networkDisplayName } from "@/utils/network-utils";

type WalletDetailNetworkProperties = {
	network: Networks.Network;
} & TransactionDetailProperties;

export const WalletDetailNetwork = ({
	network,
	borderPosition = "top",
	...properties
}: WalletDetailNetworkProperties) => {
	const { t } = useTranslation();

	return (
		<WalletDetail
			data-testid="WalletDetailNetwork"
			label={t("TRANSACTION.CRYPTOASSET")}
			extra={
				<>
					<div className="hidden sm:block">
						<NetworkIcon network={network} size="lg" />
					</div>
					<div className="flex items-center sm:hidden">
						<NetworkIcon isCompact network={network} size="lg" noShadow />
					</div>
				</>
			}
			borderPosition={borderPosition}
			{...properties}
		>
			{networkDisplayName(network)}
		</WalletDetail>
	);
};
