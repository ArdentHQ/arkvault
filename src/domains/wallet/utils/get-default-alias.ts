import { Networks } from "@ardenthq/sdk";
import { Contracts } from "@ardenthq/sdk-profiles";
import { networkDisplayName } from "@/utils/network-utils";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${networkDisplayName(network)} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
