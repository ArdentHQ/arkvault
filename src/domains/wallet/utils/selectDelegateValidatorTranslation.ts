import { Networks } from "@ardenthq/sdk";
import { isMainsailNetwork } from "@/utils/network-utils";

interface Properties<T extends string | React.ReactNode> {
	network: Networks.Network;
	validatorStr: T;
	delegateStr: T;
}

export const selectDelegateValidatorTranslation = <T extends string | React.ReactNode>({ network, validatorStr, delegateStr }: Properties<T>): T => isMainsailNetwork(network) ? validatorStr : delegateStr;



