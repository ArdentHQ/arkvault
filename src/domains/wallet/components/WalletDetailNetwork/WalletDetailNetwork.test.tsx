import React from "react";

import { WalletDetailNetwork } from "./WalletDetailNetwork";
import { translations } from "@/domains/transaction/i18n";
import { availableNetworksMock } from "@/tests/mocks/networks";
import { render } from "@/utils/testing-library";

describe("WalletDetailNetwork", () => {
	it("should render", () => {
		const network = availableNetworksMock.find((network) => network.id() === "ark.devnet");

		const { container } = render(<WalletDetailNetwork network={network!} />);

		expect(container).toHaveTextContent(translations.CRYPTOASSET);

		// eslint-disable-next-line testing-library/no-node-access
		expect(container.querySelector("svg#ark")).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
