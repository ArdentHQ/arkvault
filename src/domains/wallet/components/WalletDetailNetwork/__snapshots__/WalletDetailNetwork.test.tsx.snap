// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletDetailNetwork should render 1`] = `
<div>
  <div
    class="no-ligatures css-19rivvh"
    data-testid="WalletDetailNetwork"
  >
    <div
      class="flex w-full items-center justify-between space-x-4"
    >
      <div
        class="w-40 space-y-2 whitespace-nowrap sm:flex-1"
      >
        <div
          class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Cryptoasset
        </div>
        <div
          class="hidden items-center font-semibold sm:flex"
        >
          ARK Devnet
        </div>
      </div>
      <div
        class="flex w-full items-center justify-end overflow-auto sm:w-auto"
      >
        <div
          class="flex w-full items-center justify-end overflow-auto pr-4 font-semibold sm:hidden"
        >
          <div
            class="w-full truncate"
          >
            ARK Devnet
          </div>
        </div>
        <div
          class="hidden sm:block"
        >
          <div
            aria-label="ARK Devnet"
            class="text-theme-secondary-700 border-theme-secondary-300 dark:border-theme-secondary-700 ring-theme-background duration-100 css-14jhlj1"
            data-testid="NetworkIcon-ARK-ark.devnet"
          >
            <div
              class="css-1i4b0eq"
              data-testid="NetworkIcon__icon"
              height="20"
              width="20"
            >
              <svg>
                ark.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          class="flex items-center sm:hidden"
        >
          <div
            aria-label="ARK Devnet"
            class="inline-flex h-5 w-5 items-center justify-center text-theme-secondary-700 border-theme-secondary-300 dark:border-theme-secondary-700"
            data-testid="NetworkIcon-ARK-ark.devnet"
          >
            <div
              class="css-1i4b0eq"
              data-testid="NetworkIcon__icon"
              height="20"
              width="20"
            >
              <svg>
                ark.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
